datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedPartitionAssigner.java

```diff
@@ -124,4 +139,28 @@
 
     return newAssignments;
   }
+
+  private void validatePartitionCountAndThrow(String datastream, int numTasks, int numPartitions,
+      int maxPartitionsPerTask) {
+    // conversion to long to avoid integer overflow
+    if (numTasks * (long) maxPartitionsPerTask < numPartitions) {
```

Comment:
if `maxPartitionsPerTask` is set to `integer.MAX_VALUE`, will this work? Is there a possibility of hitting the limits.

Reply:
Integer.MAX_VALUE squared is ~ 4.61 * 10^18. It's smaller than Long.MAX_VALUE. There's no possibility of overflow if we do 64bit arithmetics here.